docs/decisions/0002-react-i18n.rst

```diff
@@ -0,0 +1,70 @@
+2. ``react-intl``
+-----------------
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+We have business needs to translate or otherwise internationalize
+strings of text used within ``paragon``
+components. Internationalization is critical to the goal of extending
+access to education globally, a core aspect of the mission of the
+``edx`` platform.
+
+The React community has demonstrated fairly widespread adoption of
+react-intl_ for addressing this need in various contexts.
+The ``edx/studio-frontend`` project has used react-intl_ for some
+time.
+
+Decision
+--------
+
+We will adopt the use of react-intl_ for all new learner- and
+educator-facing web development using React.
+We will attempt to standardize our use of internationalization
+libraries (including but not limited to react-intl_) for existing
+React-based web development.
+
+Consequences
+------------
+
+react-intl_ uses the `React context API`_ when using the
+``FormattedMessage``, ``React.Component``-based approach to using the
+API. Because of this reliance on context, components which are passed
```

Comment:
This paragraph is kind of confusing to me. Would it be possible to get an example of what `must be rendered in
the same React component hierarchy in each of the corresponding
components` would look like, more concretely?

Reply:
I hear that. It's a little abstract, because I'm trying to phrase fundamentally negative guidance as if it were positive (i.e. I'm trying to say "do X!" instead of "don't do Y!"). Let's brainstorm a good alternative during tomorrow's FedX meeting.